### PR TITLE
EIP1-3816 / EIP1-3819 - introduced config and service for AED Explainer Document

### PR DESCRIPTION
**Pull request by NathanRussellValtech**
This PR introduces the config, template PDFs and service for generating the AED Explainer Document

There has been a fair bit of refactoring here to be able to re-use the classes etc that were setup for Temporary Certificates (rather than copy and paste them)
It might be easier to check out the branch and look at it in your IDE, but happy to go through it on a call if necessary

### CODE REVIEW

File: src/main/kotlin/uk/gov/dluhc/printapi/config/ElectorDocumentPdfTemplateProperties.kt [hunk 1]
```diff
@@ -5,7 +5,10 @@ import org.springframework.boot.context.properties.ConstructorBinding
 
 @ConfigurationProperties(prefix = "temporary-certificate.certificate-pdf")
 @ConstructorBinding
-data class TemporaryCertificatePdfTemplateProperties(
+class TemporaryCertificatePdfTemplateProperties(english: English, welsh: Welsh) :
+    ElectorDocumentPdfTemplateProperties(english, welsh)
+
+abstract class ElectorDocumentPdfTemplateProperties(
```
Comment (NathanRussellValtech) [inline src/main/kotlin/uk/gov/dluhc/printapi/config/ElectorDocumentPdfTemplateProperties.kt:11]: Came to a [general conscientious on slack](https://valtechhub.slack.com/archives/C03G86348N9/p1676452560461249) that the common term for the 3 types of certificate document (VAC, Temp Cert and AED) is "Elector Document"

File: src/main/kotlin/uk/gov/dluhc/printapi/config/PdfConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    ): ExplainerPdfTemplateDetailsFactory =
+        ExplainerPdfTemplateDetailsFactory(explainerPdfTemplateProperties)
+
+    @Bean("anonymousElectorDocumentExplainerExplainerPdfService")
+    fun anonymousElectorDocumentExplainerExplainerPdfService(
+        eroClient: ElectoralRegistrationOfficeManagementApiClient,
+        pdfFactory: PdfFactory,
+        @Qualifier("anonymousElectorDocumentExplainerPdfTemplateDetailsFactory") explainerPdfTemplateDetailsFactory: ExplainerPdfTemplateDetailsFactory
+    ): ExplainerPdfService =
+        ExplainerPdfService(eroClient, explainerPdfTemplateDetailsFactory, pdfFactory)
```
Comment (Neil-Massey) [inline src/main/kotlin/uk/gov/dluhc/printapi/config/PdfConfiguration.kt:49]: Nice reuse 👍

**Review by Neil-Massey — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/dluhc/printapi/config/ExplainerPdfTemplateProperties.kt [hunk 3]
```diff
@@ -5,9 +5,17 @@ import org.springframework.boot.context.properties.ConstructorBinding
...
 @ConfigurationProperties(prefix = "temporary-certificate.explainer-pdf")
 @ConstructorBinding
-class TemporaryCertificateExplainerPdfTemplateProperties(
+class TemporaryCertificateExplainerPdfTemplateProperties(english: English, welsh: Welsh) :
+    ExplainerPdfTemplateProperties(english, welsh)
+
+@ConfigurationProperties(prefix = "anonymous-elector-document.explainer-pdf")
+@ConstructorBinding
+class AnonymousElectorDocumentExplainerPdfTemplateProperties(english: English, welsh: Welsh) :
+    ExplainerPdfTemplateProperties(english, welsh)
```
Comment (Neil-Massey) [inline src/main/kotlin/uk/gov/dluhc/printapi/config/ExplainerPdfTemplateProperties.kt:14]: Good reuse 👍

**Pull request merged**